### ISSUE TITLE
[a11y]: Pressing Tab on an expanded dropdown collapses it but loses focus

### ISSUE THREAD

**Issue by saiteja-nizam** [carbon-design-system/carbon#22320]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.108.0

### Automated testing tool and ruleset

2.4.3 Focus Order

### Description

After selecting desired options in MultiSelect component on pressing tab dropdown is getting collapsed but focus is getting on the component.

### WCAG 2.1 Violation

2.4.3 Focus Order

### Reproduction/example

https://stackblitz.com/edit/anuggxqk?file=src%2FApp.jsx

### Steps to reproduce

select some options from `MultiSelect` dropdown options and press tab. 
dropdown will be collapsed but focus is getting on the component.

https://github.com/user-attachments/assets/e718257e-9ced-4651-9e7a-793bb1f80814

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
